### PR TITLE
Charts failing because of the feature flags not enabled

### PR DESCRIPTION
**Pull request by nik0811**
### Description of the change

In a mixed version cluster (e.g. some versions are 3.7.x and some are 3.8.x) some nodes will support a different set of features, behave differently in certain scenarios, and otherwise not act exactly the same: they are different versions after all.

Feature flags are a mechanism that controls what features are considered to be enabled or available on all cluster nodes. If a feature flag is enabled, so is its associated feature (or behavior). If not then all nodes in the cluster will disable the feature (behavior).

The feature flag subsystem allows RabbitMQ nodes with different versions to determine if they are compatible and then communicate together, regardless of their version.

This subsystem was introduced in RabbitMQ 3.8.0 to allow [rolling upgrades](https://www.rabbitmq.com/upgrade.html#rolling-upgrades) of cluster members without shutting down the entire cluster.

### Applicable issues

  - fixes #

### Checklist

- [ ] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [ ] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [ ] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [ ] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Comment by carrodher** [comment]
Thanks for your contribution, please note there are two actions failing:
- https://github.com/bitnami/charts/actions/runs/3932904004/jobs/6726016149, because you need to bump the chart version in the _Chart.yaml_
- https://github.com/bitnami/charts/actions/runs/3932904013/jobs/6726015933, because you need to set the proper metadata for any new parameter added to the _values.yaml_, see the rest of `## @param` comments in that file

**Comment by nik0811** [comment]
> Thanks for your contribution, please note there are two actions failing:
> 
> * https://github.com/bitnami/charts/actions/runs/3932904004/jobs/6726016149, because you need to bump the chart version in the _Chart.yaml_
> * https://github.com/bitnami/charts/actions/runs/3932904013/jobs/6726015933, because you need to set the proper metadata for any new parameter added to the _values.yaml_, see the rest of `## @param` comments in that file

Added all the relevant changes, even added the flags in README, still pipeline failing ?

File: bitnami/rabbitmq/values.yaml [hunk 1]
```diff
@@ -448,6 +448,12 @@ advancedConfiguration: |-
 ##
 advancedConfigurationExistingSecret: ""
 
+## This subsystem was introduced in RabbitMQ 3.8.0 to allow rolling upgrades of cluster members without shutting down the entire cluster.
+## Feature flags are a mechanism that controls what features are considered to be enabled or available on all cluster nodes. If a feature flag is enabled, so is its associated feature (or behavior). If not then all nodes in the cluster will disable the feature (behavior).
+## e.g, drop_unroutable_metric,empty_basic_get_metric,implicit_default_bindings,maintenance_mode_status,quorum_queue,virtual_host_metadata
+# @param feature_flags that controls what features are considered to be enabled or available on all cluster nodes.
+feature_flags:
```
Comment (marcosbc) [inline bitnami/rabbitmq/values.yaml:455]: We are currently using camelCase for Helm chart parameters. Could you change the parameter name to `featureFlags`?

File: bitnami/rabbitmq/README.md [hunk 2]
```diff
@@ -111,6 +111,7 @@ The command removes all the Kubernetes components associated with the chart and
 | `auth.tls.overrideCaCertificate`      | Existing secret with certificate content be mounted instead of the `ca.crt` coming from caCertificate or existingSecret/existingSecretFullChain.                        | `""`                                              |
 | `logs`                                | Path of the RabbitMQ server's Erlang log file. Value for the `RABBITMQ_LOGS` environment variable                                                                       | `-`                                               |
 | `ulimitNofiles`                       | RabbitMQ Max File Descriptors                                                                                                                                           | `65536`                                           |
+| `feature_flags`                       | RabbitMQ controls what features are considered to be enabled or available on all cluster nodes. | `drop_unroutable_metric,empty_basic_get_metric,implicit_default_bindings,maintenance_mode_status,quorum_queue,virtual_host_metadata`
```
Comment (marcosbc) [inline bitnami/rabbitmq/README.md:114]: In the README, the default value is empty. Could you check this?

File: bitnami/rabbitmq/Chart.yaml [hunk 3]
```diff
@@ -23,4 +23,4 @@ name: rabbitmq
 sources:
   - https://github.com/bitnami/containers/tree/main/bitnami/rabbitmq
   - https://www.rabbitmq.com
-version: 11.4.0
+version: 11.4.1
```
Comment (marcosbc) [inline bitnami/rabbitmq/Chart.yaml:26]: Since this is adding a new feature, we should be increasing the minor (e.g. to 11.5.0).

**Review by marcosbc — CHANGES_REQUESTED** [review]
Thanks for the contribution! Please have a look at my comments whenever you have a chance.

File: bitnami/rabbitmq/values.yaml (hunk 1 above)
Comment (nik0811) [reply bitnami/rabbitmq/values.yaml]: fixed

File: bitnami/rabbitmq/Chart.yaml (hunk 3 above)
Comment (nik0811) [reply bitnami/rabbitmq/Chart.yaml]: fixed

File: bitnami/rabbitmq/README.md (hunk 2 above)
Comment (nik0811) [reply bitnami/rabbitmq/README.md]: it is there

File: bitnami/rabbitmq/metadata [hunk 4]
```diff
@@ -0,0 +1,3 @@
+| Name      | Description             | Default        |
```
Comment (marcosbc) [inline bitnami/rabbitmq/metadata:1]: This seems like a file created by mistake. Could you remove it?

File: bitnami/rabbitmq/values.yaml [hunk 5]
```diff
@@ -448,6 +448,12 @@ advancedConfiguration: |-
 ##
 advancedConfigurationExistingSecret: ""
 
+## This subsystem was introduced in RabbitMQ 3.8.0 to allow rolling upgrades of cluster members without shutting down the entire cluster.
+## Feature flags are a mechanism that controls what features are considered to be enabled or available on all cluster nodes. If a feature flag is enabled, so is its associated feature (or behavior). If not then all nodes in the cluster will disable the feature (behavior).
+## e.g, drop_unroutable_metric,empty_basic_get_metric,implicit_default_bindings,maintenance_mode_status,quorum_queue,virtual_host_metadata
+# @param featureFlags that controls what features are considered to be enabled or available on all cluster nodes.
```
Comment (marcosbc) [inline bitnami/rabbitmq/values.yaml:454]: I'm seeing the following error in the `readme-generator-for-helm` action step:

```
######
The following errors must be fixed before proceeding
######

ERROR: Missing metadata for key: featureFlags
```

Could you check it? I believe it shoudl be like this:

```suggestion
## @param featureFlags that controls what features are considered to be enabled or available on all cluster nodes.
##
```

**Review by marcosbc — CHANGES_REQUESTED** [review]

File: bitnami/rabbitmq/metadata (hunk 4 above)
Comment (nik0811) [reply bitnami/rabbitmq/metadata]: done

File: bitnami/rabbitmq/values.yaml (hunk 5 above)
Comment (nik0811) [reply bitnami/rabbitmq/values.yaml]: done

**Comment by marcosbc** [comment]
It seems like the `update-readme-metadata` action is still failing. I believe it is because the default value is set to no value, but for it to work, you would need to set it to `""`. Could you change it?

**Comment by nik0811** [comment]
> It seems like the `update-readme-metadata` action is still failing. I believe it is because the default value is set to no value, but for it to work, you would need to set it to `""`. Could you change it?

Screenshot 2023-01-25 at 6 08 45 PM
check this out

**Comment by marcosbc** [comment]
Hi @nik0811, it seems like an unrelated WordPress change was included in the MR, could you remove it?

In addition, it seems like the change increasing the minor version was reverted, and now it's only increasing the patch version. Could you check this as well?

**Comment by nik0811** [comment]
> Hi @nik0811, it seems like an unrelated WordPress change was included in the MR, could you remove it?
> 
> In addition, it seems like the change increasing the minor version was reverted, and now it's only increasing the patch version. Could you check this as well?

fixed @marcosbc

**Review by marcosbc — APPROVED** [review]
LGTM, thanks!

**Pull request merged**